FlowCrypt/Controllers/Msg/MessageViewController.swift

```diff
@@ -312,95 +261,56 @@ extension MessageViewController {
 
 // MARK: - Handle Actions
 
-extension MessageViewController {
-    @objc private func handleInfoTap() {
-        showToast("Email us at human@flowcrypt.com")
-    }
+extension MessageViewController: MessageActionsHandler {
 
-    @objc private func handleMarkUnreadTap() {
-        messageOperationsProvider.markAsUnread(message: input.objMessage, folder: input.path)
-            .then(on: .main) { [weak self] in
-                guard let self = self else { return }
-                self.input.objMessage = self.input.objMessage.markAsRead(false)
-                self.onCompletion?(MessageAction.changeReadFlag, self.input.objMessage)
-                self.navigationController?.popViewController(animated: true)
-            }
-            .catch(on: .main) { [weak self] error in
-                self?.showToast("Could not mark message as unread: \(error)")
-            }
-    }
-
-    @objc private func handleAttachmentTap() {
-        showToast("Downloading attachments is not implemented yet")
-    }
-
-    @objc private func handleTrashTap() {
-        showSpinner()
-
-        trashFolderProvider.getTrashFolderPath()
-            .then { [weak self] trashPath in
-                guard let strongSelf = self, let path = trashPath else {
-                    self?.permanentlyDelete()
-                    return
-                }
-
-                strongSelf.input.path == trashPath
-                    ? strongSelf.permanentlyDelete()
-                    : strongSelf.moveToTrash(with: path)
-            }
-            .catch(on: .main) { error in
-                self.showToast(error.localizedDescription)
+    func handleMarkUnreadTap() {
+        Task {
+            do {
+                try await messageOperationsProvider.markAsUnread(message: input.objMessage, folder: input.path)
+                onCompletion(MessageAction.markAsRead(false), .init(message: self.input.objMessage))
+                navigationController?.popViewController(animated: true)
+            } catch {
+                showToast("Could not mark message as unread: \(error)")
```

Comment:
maybe `handleOpErr(operation: .markUnread)` for consistency?

Reply:
Better to ask @Kharchevskyi

Reply:
He's not available for the next few days. Can file an issue for later.